service/dap/server.go

```diff
@@ -742,7 +747,57 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
-	if mode == "debug" || mode == "test" {
+	// On replay mode, validate its required parameters to define the debugger
+	// Backend and CoreFile properties
+	if mode == "replay" {
+		// Check against replay parameters and decide the backend based on which one was received
+		coreDumpPath, _ := request.Arguments["coreDumpPath"].(string)
+		traceDirectory, _ := request.Arguments["traceDirectory"].(string)
+
+		if (coreDumpPath != "") && (traceDirectory != "") {
```

Comment:
() are unnecessary. 

How about checking coreDumpPath == "" && traceDirectory == "" too here?

if (coreDumpPath != ""  && traceDirectory != "") || (coreDumpPath == "" && traceDirectory == "") {
   s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", "The 'replay' mode must have either a  coreDumpPath or a traceDirectory, but not both." )
   return
 }
 
 If we take a path to let users specify 'backend' in the launch configuration, we will need to also check whether that is consistent when specified.

Reply:
Done